ray/tests/test_e2e.py

```diff
@@ -0,0 +1,33 @@
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import pytest
+
+from datadog_checks.base import AgentCheck
+from datadog_checks.dev.utils import get_metadata_metrics
+
+from .common import HEAD_INSTANCE, HEAD_METRICS, OPTIONAL_METRICS, WORKER1_INSTANCE, WORKER_METRICS
+
+pytestmark = [pytest.mark.e2e, pytest.mark.usefixtures("dd_environment")]
+
+
+@pytest.mark.parametrize(
+    'instance, metrics',
+    [
+        pytest.param(HEAD_INSTANCE, HEAD_METRICS, id='head'),
+        pytest.param(WORKER1_INSTANCE, WORKER_METRICS, id='worker'),
+    ],
+)
+def test_check(dd_agent_check, instance, metrics):
+    aggregator = dd_agent_check(instance, rate=True)
+
+    for expected_metric in metrics:
+        if expected_metric in OPTIONAL_METRICS:
+            aggregator.assert_metric(f"ray.{expected_metric}", at_least=0)
```

Comment:
We assert them because otherwise we can't `assert_all_metrics_covered` and if they are collected, `assert_metrics_using_metadata` will also validate their metadata. Only if they are collected, but still it can spot some issues

Reply:
Oh, good point! TIL, thanks!